projects/toggle-lib/src/lib/components/toggle/toggle.component.ts

```diff
@@ -18,46 +20,40 @@ let counter = 0;
   providers: [
     {
       provide: NG_VALUE_ACCESSOR,
-      useExisting: forwardRef(() => ToggleComponent),
+      useExisting: forwardRef(() => NextToggleComponent),
       multi: true,
     },
   ],
 })
-export class ToggleComponent implements ControlValueAccessor, OnInit {
-  @Input() disabled: boolean;
-  @Input() required: boolean;
-  @Input() tabIndex: number;
-  @Input() id: string;
+export class NextToggleComponent implements ControlValueAccessor, AfterViewInit {
+  @Input() public disabled: boolean;
+  @Input() public required: boolean;
+  @Input() public tabIndex: number;
+  @Input() public id = `next-toggle-${++counter}`;
 
-  inputId: string;
+  private isChecked = false;
 
-  private inputIdGenerator = this.idGenerator('on-off-checkbox', 'input');
-  public controlValueAccessorChangeFn: (value: any) => void;
-  protected onTouched: (value: any) => void;
-
-  private _checked = false;
-
-  constructor(private changeDetectorRef: ChangeDetectorRef) {}
-
-  ngOnInit(): void {
-    if (this.id) {
-      this.inputId = this.id;
-    } else {
-      this.inputId = this.inputIdGenerator();
-    }
+  public get inputId(): string {
+    return this.id;
   }
 
-  get checked(): any {
-    return this._checked;
+  public get checked(): any {
+    return this.isChecked;
   }
 
-  set checked(checked: any) {
+  public set checked(checked: any) {
     if (checked !== this.checked) {
-      this._checked = checked;
+      this.isChecked = checked;
       this.changeDetectorRef.markForCheck();
     }
   }
 
+  constructor(private changeDetectorRef: ChangeDetectorRef, private renderer: Renderer2, private el: ElementRef) {}
+
+  public ngAfterViewInit() {
+    this.renderer.removeAttribute(this.el.nativeElement, 'id');
```

Comment:
Why we should do it?

Reply:
if you do it this way you can set `id="string"` instead of `[id]="'string'"`